### PR TITLE
React/PT-131/Android accessibility

### PR DESCRIPTION
**Pull request by danybony**
This PR investigates the following aspects of accessibility:

- Add a custom text read from TalkBack instead of the default one
- Ignore certain elements of a group when navigating (for example in a list)

Also uses the Autolink library to automatically highlight links inside a text and handle click.
This is not working with accessibility though (it's not possible to select the link inside the text when navigating with talkback on). An issue will be opened to the library repo.

image

Paired with ☕

### CODE REVIEW

File: ReactNative/ReactTwitter/core/views/common/avatar-view.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ class AvatarView extends Image {
     return <Image
       source= {source}
       style = {style.image}
+      importantForAccessibility='no'
```
Comment (danybony) [inline ReactNative/ReactTwitter/core/views/common/avatar-view.js]: The image is ignored when TalkBack is on

File: ReactNative/ReactTwitter/core/views/tweetsListItem.js [hunk 2]
```diff
@@ -36,10 +37,10 @@ var TweetsListItem = React.createClass({
             <View style={styles.textContainer}>
               <View style={styles.userInfoContainer}>
                 <Text style={styles.tweet_author}>{this.props.author_name}</Text>
-                <Text style={styles.tweet_author_handle}>@{this.props.author_handle}</Text>
+                <Text importantForAccessibility='no' style={styles.tweet_author_handle}>@{this.props.author_handle}</Text>
```
Comment (danybony) [inline ReactNative/ReactTwitter/core/views/tweetsListItem.js]: Ignore the user handle when navigating between the list items

File: ReactNative/ReactTwitter/core/views/tweetsListItem.js [hunk 3]
```diff
@@ -36,10 +37,10 @@ var TweetsListItem = React.createClass({
...
               </View>
               <Text style={styles.tweet_time}>{formattedTime}</Text>
-              <Text style={styles.tweet_text}>{this.props.text}</Text>
+              <Text style={styles.tweet_text} accessibilityLabel={textWithoutUrls}>{this.props.text}</Text>
```
Comment (danybony) [inline ReactNative/ReactTwitter/core/views/tweetsListItem.js]: Replace the read text (not the displayed one) with one without urls

**Comment by danybony** [comment]
It would be good to have a feedback from @ataulm as well, if we could easily test something else

File: ReactNative/ReactTwitter/core/views/tweetsListItem.js (hunk 2 above)
Comment (ataulm) [inline ReactNative/ReactTwitter/core/views/tweetsListItem.js]: do these work just by mapping to the xml attrs?

File: ReactNative/ReactTwitter/core/views/tweetView.js [hunk 4]
```diff
@@ -8,9 +8,10 @@ import {
   Navigator
 } from 'react-native'
 import AndroidBackNavigationMixin from './mixins/android-back-navigation'
-var novodaDataFormat = require('./common/data-format.js')
-var TwitterRequestsService = require('../service/twitter-requests-service.js')
-var AvatarView = require('./common/avatar-view.js')
+const novodaDataFormat = require('./common/data-format.js')
```
Comment (ataulm) [inline ReactNative/ReactTwitter/core/views/tweetView.js]: shoudl this be `NovodaDataFormat`?

**Comment by ataulm** [comment]
:+1:

**Comment by ataulm** [comment]
Is it easy to determine if talkback is enabled and present a different UI instead?

File: ReactNative/ReactTwitter/core/views/tweetsListItem.js (hunk 2 above)
Comment (danybony) [inline ReactNative/ReactTwitter/core/views/tweetsListItem.js]: yes exactly, these attributes are passes as properties for the component constructor, in this case `Text`

File: ReactNative/ReactTwitter/core/views/tweetView.js (hunk 4 above)
Comment (danybony) [inline ReactNative/ReactTwitter/core/views/tweetView.js]: 👍  good catch

**Comment by danybony** [comment]
@ataulm I haven't find that.
It should be possible creating a specific Android component, but haven't had time for that

**Pull request merged**